backend/FwLite/FwLiteShared/Projects/CombinedProjectsService.cs

```diff
@@ -143,12 +155,38 @@ private ProjectRole FromRole(UserProjectRole role) =>
             _ => ProjectRole.Unknown
         };
 
-    public async Task DownloadProject(string code, LexboxServer server)
+    [JSInvokable]
+    public async Task<DownloadProjectByCodeResult> DownloadProjectByCode(string code, LexboxServer server, string? userRole = null)
```

Comment:
let's use the user role enum here

Reply:
Ah, that's what it's for: a string-serialized version of the enum. I did wonder about that, but given the name I thought it was connected to the Role property of the ProjectUsers object. I'll make that change.